lib/script/script.js

```diff
@@ -713,14 +727,44 @@ Script.fromAddress = function(address) {
 
 /**
  * @param {Network=} network
- * @return {Address} the associated address for this script
+ * @return {Address|boolean} the associated address information object
+ * for this script if any, or false
+ */
+Script.prototype.getAddressInfo = function() {
+  var Address = require('../address');
+  var info = {};
+  if (this.isScriptHashOut()) {
+    info.hashBuffer = this.getData();
+    info.type = Address.PayToScriptHash;
+  } else if (this.isPublicKeyHashOut()) {
+    info.hashBuffer = this.getData();
+    info.type = Address.PayToPublicKeyHash;
+  } else if (this.isPublicKeyHashIn()) {
+    // hash the publickey found in the scriptSig
+    info.hashBuffer = Hash.sha256ripemd160(this.chunks[1].buf);
+    info.type = Address.PayToPublicKeyHash;
+  } else if (this.isScriptHashIn()) {
+    // hash the redeemscript found at the end of the scriptSig
+    info.hashBuffer = Hash.sha256ripemd160(this.chunks[this.chunks.length - 1].buf);
```

Comment:
same here... a p2sh input script has a LOT more data than the hash of the redeem script... why would `getData` return that?